tests/test_editable.py

```diff
@@ -0,0 +1,189 @@
+# SPDX-FileCopyrightText: 2021 The meson-python developers
+#
+# SPDX-License-Identifier: MIT
+
+import os
+import pathlib
+import sys
+
+import pytest
+
+import mesonpy
+
+from mesonpy import _editable
+
+from .test_wheel import EXT_SUFFIX
+
+
+def test_walk():
+    entries = set(_editable.walk(os.fspath(pathlib.Path(__file__).parent.joinpath('packages/complex/complex'))))
+    assert entries == {
+        pathlib.Path('__init__.py'),
+        pathlib.Path('more/__init__.py'),
+        pathlib.Path('namespace/bar.py'),
+        pathlib.Path('namespace/foo.py')
+    }
+
+
+def test_nodes_tree():
+    tree = _editable.Node()
+    tree[('aa', 'bb', 'cc')] = 'path1'
+    tree[('aa', 'dd')] = 'path2'
+    assert tree['aa']['bb']['cc'] == 'path1'
+    assert tree['aa']['dd'] == 'path2'
+    assert tree[('aa', 'bb', 'cc')] == 'path1'
+    assert tree.get(('aa', 'bb', 'cc')) == 'path1'
+    assert tree['aa'] == tree[('aa', )]
+    assert tree.get(('aa', 'gg')) is None
+    assert tree[('aa', 'gg')] == _editable.Node()
+
+
+def test_collect():
+    root = os.fspath(pathlib.Path(__file__).parent.joinpath('packages/complex'))
```

Comment:
> Also, since you are using `joinpath`, why are you still assuming the path separator on the parts to join?

Because it is how `pathlib` works:
```
>>> import pathlib
>>> pathlib.PureWindowsPath('foo').joinpath('bar/baz')
PureWindowsPath('foo/bar/baz')
```